### ISSUE TITLE
`hardware.pulseaudio.enable` defaults to false, but I get assertion error if I remove the explicitly-set option from configuration file.

### ISSUE THREAD

**Issue by Ashvith10** [NixOS/nixpkgs#261320]
### Describe the bug
In the [NixOS options page](https://search.nixos.org/options), it mentions that `hardware.pulseaudio.enable` is false by default, yet I am supposed to include this option explicitly, or else the rebuild fails.

### Steps To Reproduce
1. Remove `hardware.pulseaudio.enable` from the configuration file
2. Try rebuilding the configuration
3. This error shows up:
    > Failed assertions:
       - Using PipeWire as the sound server conflicts with PulseAudio. This option requires `hardware.pulseaudio.enable` to be set to false

### Expected behavior
Should build without any error

### Screenshots
\-

### Additional context
\-

### Notify maintainers
@Mic92 
@pennae 
@wegank 
@Yarny0 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.56, NixOS, 23.11 (Tapir), 23.11pre534806.5e4c2ada4fcd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(ashvith): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

**Comment by Yarny0** [comment]
If you are not explicitely enabling pulseaudio, likely something else does.

A lot of desktop environments seem to enable pulseaudio by default:

```
$ git grep -l 'pulseaudio.enable = mkDefault true'
nixos/modules/services/x11/desktop-managers/budgie.nix
nixos/modules/services/x11/desktop-managers/cinnamon.nix
nixos/modules/services/x11/desktop-managers/deepin.nix
nixos/modules/services/x11/desktop-managers/gnome.nix
nixos/modules/services/x11/desktop-managers/pantheon.nix
```

If pipewire can act as substitute of pulseaudio, maybe it would be possible to generalise these settings such that pulseaudio is enabled only if pipewire isn't.  On the other hand, I'm not sure whether desktop environemnts should care about the conflict of pulseaudio and pipewire.

**Comment by Ashvith10** [comment]
Since I'm on GNOME, I'm assuming that `nixos/modules/services/x11/desktop-managers/gnome.nix` is enabling this. But wouldn't it be for the best to switch to pipewire?

**Comment by genevieve-me** [comment]
Could we reopen this? It's a bit strange getting bitten by this when trying out GNOME, as far as I can tell nothing in the GNOME module relies on pulseaudio vs pipewire.

**Comment by Ashvith10** [comment]
@genevieve-me You can use Pipewire by:
1. disabling pulseaudio
2. enabling pipewire and then pipewire-pulseaudio, so that GNOME does not complain anymore about missing pulseaudio

I don't use NixOS anymore, so I may be in the wrong here somewhere. It looks like [GNOME still uses pulseaudio](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/x11/desktop-managers/gnome.nix#L323).

**Comment by genevieve-me** [comment]
Yes, I just set `hardware.pulseaudio.enable = lib.mkForce false;`, but I don't see any reason why this should be necessary. I could open a new issue to remove that line, but it could be tracked here (yarny0 mentioned  the rationale to remove it above).
